server/jetstream_cluster.go

```diff
@@ -5333,6 +5333,30 @@ func (js *jetStream) stopUpdatesSub() {
 	}
 }
 
+func (s *Server) sendDomainLeaderElectAdvisory() {
+	js, cc := s.getJetStreamCluster()
+	if js == nil || cc == nil {
+		return
+	}
+
+	js.mu.RLock()
+	node := cc.meta
+	js.mu.RUnlock()
+
+	adv := &JSDomainLeaderElectedAdvisory{
+		TypedEvent: TypedEvent{
+			Type: JSDomainLeaderElectedAdvisoryType,
+			ID:   nuid.Next(),
+			Time: time.Now().UTC(),
```

Comment:
Should this time stamp be obtained while the lock is held, so that we don't report that at time T the leader was in one place when it had actually already moved out, or do we go "it's second resolution, you can't rely on that for anything but informational purposes in the context of subscribing to a complete stream"?